File: 백준/Silver/2559. 수열/수열.cc
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL); cout.tie(NULL);

    int N, K, num, ret=-INT_MAX;
    cin >> N >> K;

    int psum[N];
    memset(psum, 0, sizeof(psum));
    for (int i=1; i<=N; i++) {
        cin >> num;
        psum[i] = psum[i-1] + num;
    }

    for (int i=K; i<=N; i++) {
        ret = max(ret, psum[i] - psum[i-K]);
    }
    cout << ret << '\n';
    
    return 0;
}

File: 백준/Silver/2559. 수열/수열.py
N, K = map(int, input().split())
arr = [0] + list(map(int, input().split()))

ret = 100000 * -100
psum = [0] * (N+1)

for i in range(1, N+1):
    psum[i] += psum[i-1] + arr[i]

for i in range(K, N+1):
    ret = max(ret, psum[i] -  psum[i-K])

print(ret)